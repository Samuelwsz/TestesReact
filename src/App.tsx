import "./styles/global.css"
import Header from "./components/header"
import GameStart from "./components/gameStart"
import { RouterProvider } from "react-router-dom"
import { router } from "./routes/routes"

function App() {
  return (
    <>
      <Header />

      <GameStart />
    </>
  )
}

export default App
