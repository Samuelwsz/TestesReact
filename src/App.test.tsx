import { render, screen } from "@testing-library/react"
import App from "./App"
import { BrowserRouter } from "react-router-dom"

describe("Jest", () => {
  it("should display elements", () => {
    render(<App />)
    expect(screen.debug())
  })
})

/*Vite + React + TS*/

test("jest", () => {
  render(<App />)
  const linkElement = screen.getByText(/aaa/)
  expect(linkElement).toBeInTheDocument()
})

describe("a pagina de configuração", () => {
  test("deve ser renderizado corretamente", () => {
    const { container } = render(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
