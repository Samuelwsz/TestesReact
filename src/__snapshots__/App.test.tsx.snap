// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a pagina de configuração deve ser renderizado corretamente 1`] = `
<div>
  <div
    class="bg-blue-500 pt-12"
  >
    <div
      class="flex gap-3 justify-center flex-col items-center md:flex-row md:items-center"
    >
      <img
        alt=""
        class="w-96 h-40 md:block hidden"
        src="test-file-stub"
      />
      <img
        alt=""
        class="w-72 h-60 md:hidden block"
        src="test-file-stub"
      />
      <img
        alt=""
        class="md:mt-0 mt-4 w-96 h-80 "
        src="test-file-stub"
      />
    </div>
  </div>
</div>
`;
